index.js

```diff
@@ -1,6 +1,12 @@
 'use strict'
 
-var Module = require('module').Module
+var BuiltinModule = require('module')
+
+// Guard against poorly mocked module constructors
+var Module = module.constructor.length > 1
+  ? module.constructor
+  : BuiltinModule
+
```

Comment:
The `module.constructor.length` guards against Jest or others which mock `module` as just a plain object literal `{}`. A compliant constructor will have a `.length` of at least 2 *(id, parent)*.